packages/bigtable/src/cluster.js

```diff
@@ -0,0 +1,247 @@
+     *   // The `cluster` data has been populated.
+     * });
+     */
+    get: true,
+
+    /**
+     * Get the cluster metadata.
+     *
+     * @param {function} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {object} callback.metadata - The metadata.
+     * @param {object} callback.apiResponse - The full API response.
+     *
+     * @example
+     * cluster.getMetadata(function(err, metadata, apiResponse) {});
+     */
+    getMetadata: {
+      protoOpts: {
+        service: 'BigtableInstanceAdmin',
+        method: 'getCluster'
+      },
+      reqOpts: {
+        name: id
+      }
+    }
+  };
+
+  var config = {
+    parent: instance,
+    id: id,
+    methods: methods,
+    createMethod: function(_, options, callback) {
+      instance.createCluster(name, options, callback);
+    }
+  };
+
+  common.GrpcServiceObject.call(this, config);
+}
+
+nodeutil.inherits(Cluster, common.GrpcServiceObject);
+
+/**
+ * Maps the storage type to the proper integer.
+ *
+ * @private
+ *
+ * @param {string} type - The storage type (hdd, ssd).
+ * @return {number}
+ *
+ * @example
+ * Cluster.getStorageType_('ssd');
+ * // 1
+ */
+Cluster.getStorageType_ = function(type) {
+  var storageTypes = {
+    unspecified: 0,
+    ssd: 1,
+    hdd: 2
+  };
+
+  if (is.string(type)) {
+    type = type.toLowerCase();
+  }
+
+  return storageTypes[type] || storageTypes.unspecified;
+};
+
+/**
+ * Set the cluster metadata.
+ *
+ * See {module:bigtable/instance#createCluster} for a detailed explanation of
+ * the arguments.
+ *
+ * @param {object} metadata - Metadata object.
+ * @param {string} metadata.location - The cluster location.
+ * @param {number} metadata.nodes - Number of nodes allocated to the cluster.
+ * @param {string} metadata.storage - The cluster storage type.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {Operation} callback.operation - An operation object that can be used
+ *     to check the status of the request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * var callback = function(err, operation, apiResponse) {
+ *   if (err) {
+ *     // Error handling omitted.
+ *   }
+ *
+ *   operation
+ *     .on('error', console.error)
+ *     .on('complete', function() {
+ *       // The cluster has been successfully updated.
+ *     });
+ * };
+ *
+ * cluster.setMetadata({
+ *   location: 'us-central1-b',
+ *   nodes: 3,
+ *   storage: 'ssd'
+ * }, callback);
+ */
+Cluster.prototype.setMetadata = function(options, callback) {
+  var self = this;
+
+  var protoOpts = {
+    service: 'BigtableInstanceAdmin',
+    method: 'updateCluster'
+  };
+
+  var reqOpts = {
+    name: this.id
+  };
+
+  if (options.location) {
+    reqOpts.location = format('projects/{project}/locations/{location}', {
```

Comment:
Maybe do a quick detection to see if the user gave a full path location, i.e. only do this ^ if `options.location` doesn't include a `/`.